src/main/java/com/puppycrawl/tools/checkstyle/site/DescriptionMacro.java

```diff
@@ -0,0 +1,182 @@
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * A macro that inserts a description of module from its Javadoc.
+ */
+@Component(role = Macro.class, hint = "description")
+public class DescriptionMacro extends AbstractMacro {
+
+    /** New line escape character. */
+    private static final String NEWLINE = "\n";
+    /** A newline with 8 spaces of indentation. */
+    private static final String INDENT_LEVEL_8 = SiteUtil.getNewlineAndIndentSpaces(8);
+    /** A newline with 10 spaces of indentation. */
+    private static final String INDENT_LEVEL_10 = SiteUtil.getNewlineAndIndentSpaces(10);
+
+    @Override
+    public void execute(Sink sink, MacroRequest request) throws MacroExecutionException {
+        final Path modulePath = Paths.get((String) request.getParameter("modulePath"));
+        final String moduleName = CommonUtil.getFileNameWithoutExtension(modulePath.toString());
+
+        final Set<String> propertyNames = getPropertyNames(moduleName);
+        final Map<String, DetailNode> propertiesJavadocs = SiteUtil.getPropertiesJavadocs(
+            propertyNames, moduleName, modulePath);
+
+        final DetailNode moduleJavadoc = propertiesJavadocs.get(moduleName);
+
+        final int descriptionEndIndex = getDescriptionEndIndex(moduleJavadoc, propertyNames);
+        final String moduleDescription = JavadocMetadataScraper.constructSubTreeText(
+            moduleJavadoc, 0, descriptionEndIndex);
+
+        writeOutDescription(moduleDescription, sink);
+
+    }
+
+    /**
+     * Assigns values to each instance variable.
+     *
+     * @param moduleName name of module.
+     * @return set of property names.
+     * @throws MacroExecutionException if the module could not be retrieved.
+     */
+    private static Set<String> getPropertyNames(String moduleName)
+            throws MacroExecutionException {
+        final Object instance = SiteUtil.getModuleInstance(moduleName);
+        final Class<?> clss = instance.getClass();
+
+        return SiteUtil.getPropertiesForDocumentation(clss, instance);
+    }
+
+    /**
+     * Gets the end index of the description.
+     *
+     * @param moduleJavadoc javadoc of module.
+     * @param propertyNamesSet Set with property names.
+     * @return the end index.
+     */
+    private static int getDescriptionEndIndex(DetailNode moduleJavadoc,
+                                              Set<String> propertyNamesSet) {
+        int descriptionEndIndex = -1;
+
+        if (propertyNamesSet.isEmpty()) {
+            descriptionEndIndex += getParentSectionStartIndex(moduleJavadoc);
+        }
+        else {
+            final String somePropertyName = propertyNamesSet.iterator().next();
+
+            final Optional<DetailNode> somePropertyModuleNode =
+                SiteUtil.getPropertyJavadocNodeInModule(
+                    somePropertyName, moduleJavadoc);
```

Comment:
we get the first property of the module, it helps us to get the ending index of the description so these variables should be renamed to:

```suggestion
            final String firstPropertyName = propertyNamesSet.iterator().next();

            final Optional<DetailNode> firstPropertyModuleNode =
                SiteUtil.getPropertyJavadocNodeInModule(
                    firstPropertyName, moduleJavadoc);
```

Reply:
Ah I missed it. My bad!

Reply:
@Zopsss , please resolve conversation or type "resolved" to let us all be clear that this review item is resolved.

Reply:
Resolved 👍🏻